### ISSUE TITLE
SpecCluster __repr__ says 0 workers 4 threads when using dask-worker directly to connect to the scheduler

### ISSUE THREAD

**Issue by lesteve** [dask/distributed#3046]
This is with distributed 3ad195d:
```py
from distributed import LocalCluster
cluster = LocalCluster(n_workers=0)

import subprocess
p = subprocess.Popen(['dask-worker', cluster.scheduler_address])

import time
# to give the time to the dask worker to connect
time.sleep(1)
print(cluster)
```

Output:
```
LocalCluster('tcp://127.0.0.1:42801', workers=0, threads=4, memory=16.74 GB)
```

So 0 workers and 4 threads is quite unexpected ... I would expect 1 worker and 4 threads instead.

**Comment by mrocklin** [comment]
Ah, indeed.  The `workers=` attribute is defined by `len(self.workers)` while threads and memory are defined by `self.scheduler_info` (which is obtained by polling the scheduler).

We can easily move `workers=` to also defer to `scheduler_info`.  I'll put this on my TODO list for today.

**Comment by jrbourbeau** [comment]
Closed by #3058
